onnxruntime/core/providers/cuda/cuda_kernel.h

```diff
@@ -61,6 +53,11 @@ class CudaKernel : public OpKernel {
     return provider_->GetTransientScratchBuffer<T>(count_or_bytes);
   }
 
+  template <typename T>
+  inline IAllocatorUniquePtr<T> AllocateBufferOnCPUPinned(size_t count_or_bytes) const {
```

Comment:
We moved all its actual allocation logics to `CUDAExecutionProvider::AllocateBufferOnCPUPinned`.